sdk/opa.go

```diff
@@ -393,6 +482,48 @@ func evaluate(ctx context.Context, args evalArgs) (interface{}, ast.Value, map[s
 	return rs[0].Expressions[0].Value, inputAST, bundles, nil
 }
 
+type partialEvalArgs struct {
+	runtime  *ast.Term
+	compiler *ast.Compiler
+	store    storage.Store
+	txn      storage.Transaction
+	unknowns []string
+	query    string
+	now      time.Time
+	input    interface{}
+	m        metrics.Metrics
+}
+
+func partial(ctx context.Context, args partialEvalArgs) (*rego.PartialQueries, ast.Value, map[string]server.BundleInfo, error) {
+
+	bundles, err := bundles(ctx, args.store, args.txn)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	inputAST, err := ast.InterfaceToValue(args.input)
+	if err != nil {
+		return nil, nil, bundles, err
+	}
+	re := rego.New(
```

Comment:
Would we want to pass the print hook here to allow calling `print` in these policies?

Reply:
Seems like that would be good.  Can you point me to an example of how to implement it, I can make the changes.

Reply:
Sure! See the `evalArgs` struct in the same file, and how the print hook there is passed to the evaluator.

Reply:
Added the printhook in